wagtail/wagtailcore/tests/test_blocks.py

```diff
@@ -1271,6 +1301,71 @@ class ValidatedBlock(blocks.StreamBlock):
             3: ['Enter a valid URL.'],
         })
 
+    def test_block_level_validation_no_errors(self):
```

Comment:
@timheap, I think `test_block_level_validation_no_errors` and `test_block_level_validation_throws_errors` must not be in `TestStreamBlock`, because **they related only to `ValidatedBlock`**. If you want to be sure that `ValidationError`s raised as expected you can define `TestValidatedBlock` and put these two test in this test case. But I think this is the way to `TestTestValidatedBlock`, `TestTestTestValidatedBlock` and so on :) I think so because `ValidatedBlock` written only for test purposes. 

As **better solution in my opinion** I can suggest you:
* write test that checks html returned by `render_form` with errors (already done with `test_block_level_validation_renders_errors`)
* write test that checks html returned by `render_form` without errors
* remove `test_block_level_validation_no_errors` and `test_block_level_validation_throws_errors`

If we do so, we will still have example that closely matches real use case and tests that covers changes in code (positive and negative scenarios).

What do you think about it?